src/Components/test/E2ETest/Infrastructure/WebDriverExtensions/BasicTestAppWebDriverExtensions.cs

```diff
@@ -0,0 +1,24 @@
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+
+namespace Microsoft.AspNetCore.Components.E2ETest
+{
+    internal static class BasicTestAppWebDriverExtensions
+    {
+        public static IWebElement MountTestComponent<TComponent>(this IWebDriver browser) where TComponent : IComponent
```

Comment:
Do we need this as both an extension method and protected method on a base class?

Reply:
I took the base class methods out. WaitUntilTestSelectorReady and WaitUntilExists. I've combined them in the WaitAssert class so that they all respect the same timeouts that the other infrastructure improvements PR has.